eZ/Bundle/EzPublishCoreBundle/Command/RemoveVersionsCommand.php

```diff
@@ -0,0 +1,198 @@
+        $removedVersionsCounter = 0;
+
+        foreach ($contentIds as $contentData) {
+            try {
+                $contentInfo = $this->contentService->loadContentInfo($contentData['id']);
+                $versions = $this->contentService->loadVersions($contentInfo);
+                $versionsCount = \count($versions);
+
+                $output->writeln(sprintf(
+                    '<info>Content %d has %d version(s)</info>',
+                    $contentData['id'],
+                    $versionsCount
+                ), Output::VERBOSITY_VERBOSE);
+
+                $versions = array_slice(array_reverse($versions), (int) $input->getOption('keep'));
+
+                foreach ($versions as $version) {
+                    if (
+                        (!$version->isPublished() && $input->getOption('status') === self::VERSION_ALL) ||
+                        ($version->isDraft() && $input->getOption('status') === self::VERSION_DRAFT) ||
+                        ($version->isArchived() && $input->getOption('status') === self::VERSION_ARCHIVED)
+                    ) {
+                        $this->contentService->deleteVersion($version);
+                        ++$removedVersionsCounter;
+                        $output->writeln(sprintf(
+                            "Content's (%s) version (%s) has been deleted.",
+                            $contentInfo->id,
+                            $version->id
+                        ), Output::VERBOSITY_VERBOSE);
+                    }
+                }
+            } catch (\Exception $e) {
+                $output->writeln(sprintf(
+                    '<error>%s</error>',
+                    $e->getMessage()
+                ));
+            }
+        }
+
+        $output->writeln(sprintf(
+            '<info>Removed %d unwanted contents version(s).</info>',
+            $removedVersionsCounter
+        ));
+    }
+
+    protected function getObjectsIds(int $keep, string $status): array
+    {
+        $query = $this->connection->createQueryBuilder()
+                ->select('c.id, v.status')
+                ->from('ezcontentobject', 'c')
+                ->leftJoin('c', 'ezcontentobject_version', 'v', 'v.contentobject_id = c.id')
+                ->groupBy('c.id', 'v.status')
+                ->having('count(c.id) > :keep');
+        $query->setParameter('keep', $keep);
+
+        if ($status === self::VERSION_DRAFT || $status === self::VERSION_ARCHIVED) {
+            $query->where('v.status = :status');
+            $query->setParameter('status', $this->getVersionInfoStatus($status));
+        }
+
+        $stmt = $query->execute();
+
+        return $stmt->fetchAll(PDO::FETCH_ASSOC);
+    }
+
+    private function getVersionInfoStatus(string $status): ?int
+    {
+        if ($status === self::VERSION_ARCHIVED) {
+            return VersionInfo::STATUS_ARCHIVED;
+        }
+        if ($status === self::VERSION_DRAFT) {
+            return VersionInfo::STATUS_DRAFT;
+        }
```

Comment:
As a developer I expect that `getVersionInfoStatus` will throw `InvalidArgumentException` when the `status` parameter value could not be mapped to `VersionInfo::STATUS_*` const.